tests/src/404.js

```diff
@@ -52,20 +52,26 @@ const checkUrl = async (browser, url, fromUrl) => {
   }
 
   try {
-    let response = null;
+    let responses = [];
     const browserPage = await browser.newPage();
     let result = true;
 
+    // One response event for each file on the page
     browserPage.on('response', res => {
-      if (res.url().indexOf('favicon.ico') === -1) {
-        response = res;
+      if (res.url().indexOf('favicon.ico') !== -1) {
+        return;
       }
+      responses.push(res);
     });
 
     await browserPage.goto(url, { waitUntil: 'networkidle2' });
 
-    if (response && response.status() > 401) {
-      throw new Error(`status ${response.status()}`);
+    mainResponse = responses.find((response) => response.url() === url);
+    isResponseFailed = (response) => response && response.status() >= 400;
+    failedResponses = responses.filter((response) => isResponseFailed(response));
+    allResponsesOk = failedResponses.length === 0;
```

Comment:
Not sure if @bogue will approve these old style variable declaration (without `let`/`const`) :stuck_out_tongue_closed_eyes: